### PR TITLE
Fix - read Docker passwords from stdin

### PR DESCRIPTION
**Pull request by iyovcheva**
Signed-off-by: Ivana Yovcheva (VMware) <iyovcheva@vmware.com>

Docker CLI now mandates passwords are passed-in via stdin. This
change updates `docker login` to address the new requirement.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
[#650](https://github.com/openfaas/faas/issues/650)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ after_success:
 	echo "Travis tag: $TRAVIS_TAG"
```
Comment (alexellis) [inline .travis.yml]: Was space deleted here?

Comment (iyovcheva) [reply .travis.yml]: No, it doesn't appear in the log and `git diff` as well

```
diff --git a/.travis.yml b/.travis.yml
index e4d7508..06d0268 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -22,7 +22,7 @@ after_success:
        echo "Travis tag: $TRAVIS_TAG"
 
         docker tag $DOCKER_NS/faas-swarm:latest-dev $DOCKER_NS/faas-swarm:$TRAVIS_TAG;
-        docker login -u=$DOCKER_USERNAME -p=$DOCKER_PASSWORD;
+        echo $DOCKER_PASSWORD | docker login -u=$DOCKER_USERNAME --password-stdin;
         docker push $DOCKER_NS/faas-swarm:$TRAVIS_TAG;
 
         docker tag $DOCKER_NS/faas-swarm:latest-dev quay.io/$DOCKER_NS/faas-swarm:$TRAVIS_TAG;
```

Comment (alexellis) [reply .travis.yml]: It’s showing up as red / deleted in the UI. I’ll look in another browser

Comment (alexellis) [reply .travis.yml]: Perhaps the spacing is inconsistent on that line? Can you check?

Comment (iyovcheva) [reply .travis.yml]: This appears in the master branch:
https://github.com/openfaas/faas-swarm/blob/master/.travis.yml#L22

The spaces are now updated.

**Comment by alexellis** [comment]
In the commit message this could probably be worded as:

`Fix CI to read Docker passwords from stdin`
or
`Update CI to read Docker passwords from stdin`

Otherwise looks good to me.

**Pull request merged**